src/main/resources/org/jenkins/plugins/lockableresources/LockStep/help-maxWaiting.html

```diff
@@ -0,0 +1,11 @@
+<div>
+	<p>
+		By default builds a queued without any restriction on waiting to lock a common resource
+		Setting this parameter a limit is forced in the number of waiting builds. Any build trying to lock after
+		the limit is reached will trigger a cancel round trip which will abort the older waiting build.
+	</p>
+	<p>
+		For example, if builds #3 and #4 are waiting to lock the resource and build #5 reaches the lock step,
+		then build #3 will be cancelled in favor of #5 which is newer.
```

Comment:
Thus `CanceledCause` and associated machinery could also be deleted. `MilestoneStepExecution` would take care of all this.